### DISCUSSION TITLE
Consistent frame stutters occuring only when playing a map

### DISCUSSION THREAD

**Discussion by Memresable** [ppy/osu#20067] [Q&A] [answered]
This problem has been a thing for a very long time and it's really annoying to play with it, this is not a thing in osu!stable
These kinds of frame stutters only occur when playing a map in editor or normal modes [especially in maps that contain lots of objects ( streams for example or high bpm maps)]
I'm using 120 frame limiter

What I tried:
Skins didn't make a difference, tried with different frame limiters and different windowing modes and it's still happening
I used two GPUs and both have the same problem
I tested this on Linux and it's the exact same problem (currently on Windows 10)

PC Specs:
8GB RAM DDR3 1600MHz
Intel Core i7-5500U 2.40GHz (2 Cores 4 Threads)
Integrated GPU: Intel HD Graphics 5500 (The one is used for lazer)
Dedicated GPU: AMD Radeon R5 M330 (This one is really bad with OpenGL apps, I heard that other graphics APIs like Vulkan and DirectX are being implemented so that's a good thing)

Lazers' version: 2022-09-01

Logs:
[runtime.log](https://github.com/ppy/osu/files/9469420/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9469421/updater.log)
[database.log](https://github.com/ppy/osu/files/9469422/database.log)
[network.log](https://github.com/ppy/osu/files/9469423/network.log)
[performance.log](https://github.com/ppy/osu/files/9469425/performance.log)

![Inkedosu_2022-09-01_14-09-54_LI](https://user-images.githubusercontent.com/86978624/187900399-1a035a11-8b29-4691-98de-f5883bdaeeb0.jpg)
![Inkedosu_2022-09-01_14-02-47_LI](https://user-images.githubusercontent.com/86978624/187900404-9575c3ef-6a3e-4972-9ee5-c3942fc1664c.jpg)

**Comment by peppy** [comment, marked as answer]
Do these spikes occur when you are paused in the editor (or gameplay) doing nothing?

**Reply by Memresable** [reply]
They still occur, but barely and have less effect visually since nothing is happening
For the editor it's like almost the same as I'm playing but the spikes are less effective, it happens more often than in gameplay, in gameplay it takes like 2 or 3 seconds for each spike to occur but still less effective than actually playing

This is the editor (paused):
![editorspikes1](https://user-images.githubusercontent.com/86978624/188074821-a094d86c-c4ac-4da1-87ac-879bce68446d.PNG)

This is the gameplay (paused):
![gameplayspikes1](https://user-images.githubusercontent.com/86978624/188074898-a4cc2f7f-b23d-44c2-84c0-5c7aab5ff998.PNG)

**Reply by peppy** [reply]
I'm not sure why you are pointing arrows at the GC squares. There is no change in actual frame time there, so that should be irrelevant to your case.

Anyway thanks for the extra information. No idea what's going on here (quite possibly something external on your system adding overhead at certain intervals) but I'll get back to you if I have any leads.

**Reply by Memresable** [reply]
Hello again, apparently the latest update seems to have fixed the problem which is awesome! it's not as stuttery as before, so thanks!

**Reply by peppy** [reply]
Thanks for reporting back. That's great to hear! We still have a lot of optimisations coming so look forward to that.
